### DISCUSSION TITLE
SIGBUS Error on Airflow 2.9.1 with JDBC Connection to Hive on macOS with Apple M3 Pro

### DISCUSSION THREAD

**Discussion by drprolegend** [apache/airflow#40075] [General]
### Apache Airflow version

2.9.1

### What happened?

**Description:**
I encountered a fatal error while using Apache Airflow 2.9.1 on a Mac with an Apple M3 Pro chip. The issue occurs when I attempt to test a task that connects to Hive via JDBC.

**System Information:**
- OS: macOS
- uname: Darwin 23.5.0 Darwin Kernel Version 23.5.0: Wed May 1 20:13:18 PDT 2024; root .121.3~5/RELEASE_ARM64_T6030 arm64
- OS uptime: 0 days 9:05 hours
- Load average: 4.26 4.61 4.92

**Hardware**:
- CPU: Apple M3 Pro
- Total Cores: 11
- Cache Line Size: 128
- L1 Cache: 128 KB (instruction), 64 KB (data)
- L2 Cache: 4 MB

**Memory:**
- Physical: 18 GB
- Free: 332 MB
- Swap: 2 GB (Free: 1 GB)

**Java Information:**
- Java Version: JDK 21 and JDK 22 (tested with both)
- vm_info: Java HotSpot(TM) 64-Bit Server VM (22.0.1+8-16) for bsd-aarch64 JRE (22.0.1+8-16), built on 2024-03-12T10:33:16Z by "mach5one" with clang Apple LLVM 14.0.3 (clang-1403.0.22.14.1)

[hs_err_pid37676.log](https://github.com/user-attachments/files/15593711/hs_err_pid37676.log)

### What you think should happen instead?

The task should execute successfully without causing a JVM crash.

### How to reproduce

1. Install Apache Airflow 2.9.1 on macOS with Apple M3 Pro.
2. Configure Airflow to use a JDBC connection to Hive.
3. Execute the command: airflow tasks test <dag_id> <task_id> 

```
[2024-06-05T18:18:18.750+0200] {base.py:84} INFO - Using connection ID 'jdbc_hive2' for task execution.
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGBUS (0xa) at pc=0x000000012845912c, pid=37676, tid=259
#
# JRE version:  (22.0.1+8) (build )
# Java VM: Java HotSpot(TM) 64-Bit Server VM (22.0.1+8-16, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-aarch64)
# Problematic frame:
# V  [libjvm.dylib+0x45912c]  CodeHeap::allocate(unsigned long)+0x1a4
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /..../airflow/hs_err_pid37676.log
#
#

Process finished with exit code 134 (interrupted by signal 6:SIGABRT)

```

### Operating System

Apple M3 Pro 

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Taragolis** [comment]
This kind of error happen on JVM, not in Apache Airflow itself. More like troubleshooting rather than an issue in Airflow/Providers.

**Comment by potiuk** [comment]
Look it up at google/stackoverflow. There are some answers that indicate that this might be due to corrupted memory or full "/tmp" directory or similar - but it's up to you to search and check those, this has - as @Taragolis mentioned - nothing to do with Airflow.

**Reply by drprolegend** [reply]
I understand, but I have installed two versions of the JDK compatible with my system. Do I need to set any parameters? Setting "ulimit -c unlimited" doesn't resolve anything.

**Reply by potiuk** [reply]
No idea. You need to investigate similar issues. Can't help more wiht it.
